src/main/kotlin/com/coder/gateway/views/CoderGatewayRecentWorkspaceConnectionsView.kt

```diff
@@ -201,19 +202,25 @@ class CoderGatewayRecentWorkspaceConnectionsView(private val setContentCallback:
                             label("").resizableColumn().align(AlignX.FILL)
                         }.topGap(gap)
 
-                        connections.forEach { workspaceProjectIDE ->
-                            val enableLinks = listOf(WorkspaceStatus.STOPPED, WorkspaceStatus.CANCELED, WorkspaceStatus.FAILED, WorkspaceStatus.STARTING, WorkspaceStatus.RUNNING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
-                            val inLoadingState = listOf(WorkspaceStatus.STARTING, WorkspaceStatus.CANCELING, WorkspaceStatus.DELETING, WorkspaceStatus.STOPPING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
+                        val enableLinks = listOf(WorkspaceStatus.STOPPED, WorkspaceStatus.CANCELED, WorkspaceStatus.FAILED, WorkspaceStatus.STARTING, WorkspaceStatus.RUNNING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
+                        val inLoadingState = listOf(WorkspaceStatus.STARTING, WorkspaceStatus.CANCELING, WorkspaceStatus.DELETING, WorkspaceStatus.STOPPING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
 
+                        // We only display an API error on the first workspace rather than duplicating it on each workspace.
+                        if (deploymentError == null || showError) {
                             row {
                                 if (inLoadingState) {
                                     icon(AnimatedIcon.Default())
                                 }
+                                if (status.third != null) {
+                                    icon(status.third!!)
+                                }
```

Comment:
I low key hate that they named a function `let` 💀 TBH I think the assertion here is clearer, but if that's the kotlin way then I'm game haha. I could also do a `withoutNull` I suppose.

Reply:
`withoutNull` throws if null but here we are OK with the status potentially being null, so I think it would not quite fit here.

I think the main issue with assertions is that they bypass the safety provided to you by the language, which feels dangerous for an unclear benefit.

If you see a `!!` you need to reason about the entire code and determine that `status.third` is not going to be reassigned (and you need to do this every time the code is refactored), but we should let the machine do it for us.  :laughing:  Admittedly it is pretty trivial in this case, but still!

Reply:
Yeah I meant use withoutNull inside the if check, but you're totally right the `let` method is the best option here, I just wish they called it somewhere else lol.

Reply:
ahaha yeah and I swear I have to look it up every time because I always forget it is called `let`